extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -160,20 +166,88 @@ export class RealTimeConfigManager {
    * @visibleForTesting
    */
   maybeExecuteRealTimeConfig(customMacros, consentState) {
-    // TODO(keithwrightbos) - allow pub to override such that some callouts are
-    // still allowed.
-    if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
-        consentState == CONSENT_POLICY_STATE.UNKNOWN) {
-      return;
-    }
     if (!this.validateRtcConfig_(this.a4aElement_.element)) {
       return;
     }
+    this.consentState_ = consentState;
+    this.modifyRtcConfigForConsentStateSettings();
     customMacros = this.assignMacros(customMacros);
     this.rtcStartTime_ = Date.now();
     this.handleRtcForCustomUrls(customMacros);
     this.handleRtcForVendorUrls(customMacros);
-    return Promise.all(this.promiseArray_);
+    return this.promiseArray_.length ? Promise.all(this.promiseArray_) :
```

Comment:
I would assume that Promise.all empty array is equivalent to Promise.resolve... Why return null?

Reply:
I thought the same thing but I was having issues in testing where that was not the case

Reply:
This still surprises me... r u sure?